### PR TITLE
Assignment

### PR DESCRIPTION
**Pull request by Herro-Sama**

### CODE REVIEW

**Comment by Herro-Sama** [comment]
Worksheet_C submission

File: Worksheet_B/Pseudocode [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Open Terminal
```
Comment (edpowley) [inline Worksheet_B/Pseudocode:1]: The worksheet required you to write pseudocode for the similarity score calculation, i.e. to count the number of correct letters in the guessed word. Make sure you read the worksheet carefully in future!

File: Worksheet_B/Readme.MD [hunk 2]
```diff
@@ -0,0 +1 @@
+![Screenshot](FalloutHackFlow.pdf)
```
Comment (edpowley) [inline Worksheet_B/Readme.MD:1]: Flowchart looks good -- nice use of swimlanes. All that is missing is what happens when the player runs out of lives.

File: Worksheet_C/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Worksheet C
+
+- (a) This algorithm scans through a single list and compares every single item to check for duplicate items.
```
Comment (edpowley) [inline Worksheet_C/README.md:3]: ✅

File: Worksheet_C/README.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Worksheet C
+
+- (a) This algorithm scans through a single list and compares every single item to check for duplicate items. 
+- (b) The reason that this algorithm is quadratic, is because everytime you add a new item the algorithm has to run the new addition against every possible result in the list.
```
Comment (edpowley) [inline Worksheet_C/README.md:4]: ✅ but explanation could be clearer

File: Worksheet_C/README.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# Worksheet C
+
+- (a) This algorithm scans through a single list and compares every single item to check for duplicate items. 
+- (b) The reason that this algorithm is quadratic, is because everytime you add a new item the algorithm has to run the new addition against every possible result in the list. 
+- (c) The algorithm is still correct, as it runs previous items in the list against the new item meaning that once completed it will have compared all items but doesn't make duplicate comparisons.
```
Comment (edpowley) [inline Worksheet_C/README.md:5]: ✅

File: Worksheet_C/README.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
+# Worksheet C
+
+- (a) This algorithm scans through a single list and compares every single item to check for duplicate items. 
+- (b) The reason that this algorithm is quadratic, is because everytime you add a new item the algorithm has to run the new addition against every possible result in the list. 
+- (c) The algorithm is still correct, as it runs previous items in the list against the new item meaning that once completed it will have compared all items but doesn't make duplicate comparisons. 
+- (d) The algorithm is half as fast because it only has to make half the comparisons after removing the duplicate comparisons.
```
Comment (edpowley) [inline Worksheet_C/README.md:6]: ✅ although surely you mean twice as fast, not half as fast!

File: Worksheet_C/README.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
+# Worksheet C
+
+- (a) This algorithm scans through a single list and compares every single item to check for duplicate items. 
+- (b) The reason that this algorithm is quadratic, is because everytime you add a new item the algorithm has to run the new addition against every possible result in the list. 
+- (c) The algorithm is still correct, as it runs previous items in the list against the new item meaning that once completed it will have compared all items but doesn't make duplicate comparisons. 
+- (d) The algorithm is half as fast because it only has to make half the comparisons after removing the duplicate comparisons. 
+- (e) Yes it is still quadratic as it does n “things” and each “thing” is O(n), then the overall algorithm is O(n**2).
```
Comment (edpowley) [inline Worksheet_C/README.md:7]: ✅

File: Worksheet_C/README.md [hunk 8]
```diff
@@ -0,0 +1,11 @@
+# Worksheet C
+
+- (a) This algorithm scans through a single list and compares every single item to check for duplicate items. 
+- (b) The reason that this algorithm is quadratic, is because everytime you add a new item the algorithm has to run the new addition against every possible result in the list. 
+- (c) The algorithm is still correct, as it runs previous items in the list against the new item meaning that once completed it will have compared all items but doesn't make duplicate comparisons. 
+- (d) The algorithm is half as fast because it only has to make half the comparisons after removing the duplicate comparisons. 
+- (e) Yes it is still quadratic as it does n “things” and each “thing” is O(n), then the overall algorithm is O(n**2). 
+- (f) http://stackoverflow.com/questions/23809785/python-sorting-complexity-on-sorted-list - "Best case scenario: O(n), average and worst case O(nlogn). – Martijn Pieters".
```
Comment (edpowley) [inline Worksheet_C/README.md:8]: ✅ although Stack Overflow is not the best source -- would be better to use e.g. the official Python documentation wiki.

File: Worksheet_C/README.md [hunk 9]
```diff
@@ -0,0 +1,11 @@
+# Worksheet C
+
+- (a) This algorithm scans through a single list and compares every single item to check for duplicate items. 
+- (b) The reason that this algorithm is quadratic, is because everytime you add a new item the algorithm has to run the new addition against every possible result in the list. 
+- (c) The algorithm is still correct, as it runs previous items in the list against the new item meaning that once completed it will have compared all items but doesn't make duplicate comparisons. 
+- (d) The algorithm is half as fast because it only has to make half the comparisons after removing the duplicate comparisons. 
+- (e) Yes it is still quadratic as it does n “things” and each “thing” is O(n), then the overall algorithm is O(n**2). 
+- (f) http://stackoverflow.com/questions/23809785/python-sorting-complexity-on-sorted-list - "Best case scenario: O(n), average and worst case O(nlogn). – Martijn Pieters". 
+- (g) I think the answer would be O(nlogn) as that is the worst case scenario of the sorting function.
```
Comment (edpowley) [inline Worksheet_C/README.md:9]: Yes, but why is it the sorting function that dominates and not the for loop in lines 4-8?

File: Worksheet_C/README.md [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+
+- (a) This algorithm scans through a single list and compares every single item to check for duplicate items. 
+- (b) The reason that this algorithm is quadratic, is because everytime you add a new item the algorithm has to run the new addition against every possible result in the list. 
+- (c) The algorithm is still correct, as it runs previous items in the list against the new item meaning that once completed it will have compared all items but doesn't make duplicate comparisons. 
+- (d) The algorithm is half as fast because it only has to make half the comparisons after removing the duplicate comparisons. 
+- (e) Yes it is still quadratic as it does n “things” and each “thing” is O(n), then the overall algorithm is O(n**2). 
+- (f) http://stackoverflow.com/questions/23809785/python-sorting-complexity-on-sorted-list - "Best case scenario: O(n), average and worst case O(nlogn). – Martijn Pieters". 
+- (g) I think the answer would be O(nlogn) as that is the worst case scenario of the sorting function. 
+- (h) If the list is large, it would be faster to run the intial method as although it is slower to complete. It is faster than having to sort a large disorganised list.
```
Comment (edpowley) [inline Worksheet_C/README.md:10]: But sorting is O(n log n), which for large n is faster than O(n^2). In other words, when n gets sufficiently large, sorting the list (no matter how disorganised) can always be done faster than checking every pair of elements in the list.

File: Worksheet_C/README.md [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+- (d) The algorithm is half as fast because it only has to make half the comparisons after removing the duplicate comparisons. 
+- (e) Yes it is still quadratic as it does n “things” and each “thing” is O(n), then the overall algorithm is O(n**2). 
+- (f) http://stackoverflow.com/questions/23809785/python-sorting-complexity-on-sorted-list - "Best case scenario: O(n), average and worst case O(nlogn). – Martijn Pieters". 
+- (g) I think the answer would be O(nlogn) as that is the worst case scenario of the sorting function. 
+- (h) If the list is large, it would be faster to run the intial method as although it is slower to complete. It is faster than having to sort a large disorganised list. 
+- (i) A designer may decide to use the slower method for multiple reasons some examples of this are as follows. One: they are unsure of how to use a faster method and it was easier to do a linear search. Two: They have a long unsorted list and don't want to spend a large amount of time sorting the list so they can implement the second method.
```
Comment (edpowley) [inline Worksheet_C/README.md:11]: ✅

**Pull request closed without merging**